pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/realtime/partition/StreamPartitionAssignmentGenerator.java

```diff
@@ -0,0 +1,112 @@
+/**
+ * Copyright (C) 2014-2016 LinkedIn Corp. (pinot-core@linkedin.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.linkedin.pinot.controller.helix.core.realtime.partition;
+
+import com.linkedin.pinot.common.config.TableConfig;
+import com.linkedin.pinot.common.metadata.ZKMetadataProvider;
+import com.linkedin.pinot.controller.helix.PartitionAssignment;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.apache.helix.AccessOption;
+import org.apache.helix.ZNRecord;
+import org.apache.helix.store.zk.ZkHelixPropertyStore;
+
+
+/**
+ * Class to help generate partition assignment, given the table config, number of stream partitions,
+ * instances and all tables needing reassignment
+ */
+public class StreamPartitionAssignmentGenerator {
+
+  private ZkHelixPropertyStore<ZNRecord> _propertyStore;
+
+  public StreamPartitionAssignmentGenerator(ZkHelixPropertyStore<ZNRecord> propertyStore) {
+    _propertyStore = propertyStore;
+  }
+
+  /**
+   * Generates partition assignment for given table, given num partitions over given instances
+   */
+  public Map<String, PartitionAssignment> generatePartitionAssignment(TableConfig tableConfig, int numPartitions,
+      List<String> instanceNames, List<String> allTables) {
+
+    // get table configs for all tables in same tenant
+    List<TableConfig> allTableConfigs = new ArrayList<>(allTables.size());
+
+    // get current partition assignments for all tables in same tenant
+    Map<String, PartitionAssignment> tableNameToPartitionAssignment = new HashMap<>(allTables.size());
+
+    for (String tableName : allTables) {
+      allTableConfigs.add(getRealtimeTableConfig(tableName));
+      Map<String, List<String>> partitionToInstancesMap = getPartitionsToInstances(tableName);
+      if (partitionToInstancesMap != null) {
+        PartitionAssignment partitionAssignment = new PartitionAssignment(tableName, partitionToInstancesMap);
+        tableNameToPartitionAssignment.put(tableName, partitionAssignment);
+      }
+    }
+
+    StreamPartitionAssignmentStrategy streamPartitionAssignmentStrategy =
```

Comment:
This should return `UniformPartitionAssignmentStrategy` by default if we don't want to break the current tables. The factory should already take care of that, right?

Reply:
The factory does take care of it. Unless explicitly specified in the table config to use something else, the factory will return UniformPartitionAssignmentStrategy.